teps/0046-triggertemplate-parameter-escaping.md

```diff
@@ -0,0 +1,271 @@
+tags, and then generate with `hack/update-toc.sh`.
+-->
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+TriggerTemplates should make it easier to escape parameters that might cause
+issues when the rendered resources are being parsed before being stored in
+Kubernetes.
+
+This is a proposal for marking specific TriggerTemplate parameters as possibly
+containing data that might need to be escaped.
+
+## Motivation
+
+We receive user-data into Tekton Triggers in the form of JSON documents, often
+in the processing of these documents, we want to pass them through to tasks for
+parsing and further actions.
+
+TriggerTemplates are the way in which we pass these documents through, with
+templates for Kubernetes (K8s) resources, which are combined with a set of
+parameters, and these must be parseable as valid JSON objects.
+
+The current behaviour has changed recently, previously we had a simple "replace
+any quotation marks with an escaped version", literally:
+
+```go
+paramValue := strings.Replace(param.Value, `"`, `\"`, -1)
+```
+
+Unfortunately this replacement was made without context, which lead to issues
+like [#777 "Backslash quote in json will crash the trigger parser"](https://github.com/tektoncd/triggers/issues/777)
+
+In version 0.10.0 this feature was made optional, with users who wanted to
+retain the behaviour requiring to add an annotation to the TriggerTemplate to
+indicate that it should retain "the old escaping behaviour", this was seen as a
+stop-gap until we could understand better the requirements for escaping.
+
+A CEL function was added that allowed marshaling a JSON object as a string,
+which is somewhat useful, but is a barrier to entry for users who just want to
+pass JSON documents through to tasks securely.
+
+### Goals
+
+ * Make it easy to safely pass incoming JSON documents received by
+   EventListeners as parameters to PipelineRuns and TaskRuns, by means of
+   TriggerTemplates.
+
+### Non-Goals
+
+  * Providing for multiple forms of escaping data passing through
+    TriggerTemplates.
+
+### Use Cases (optional)
+ 
+## Requirements
+
+The basic requirement is that a JSON document including escaped quotes within a
```

Comment:
minor: a quick example would be nice (e.g. the Revert commit message from GitHub)